### PR TITLE
Log enhancement

### PR DESCRIPTION
**Pull request by JessonChan**
1.remove the dump locker for fileLogger
2.remove log package,use os.File directly(with a locker)
3.some typo fixed  
*benchmark*

benchmark

### CODE REVIEW

File: logs/file.go [hunk 1]
```diff
@@ -210,50 +236,47 @@ func (w *fileLogWriter) lines() (int, error) {
...
-			fname = filenameOnly + fmt.Sprintf(".%s.%03d%s", time.Now().Format("2006-01-02"), num, suffix)
-			_, err = os.Lstat(fname)
-		}
-		// return error if the last file checked still existed
-		if err == nil {
-			return fmt.Errorf("Rotate: Cannot find free log number to rename %s\n", w.Filename)
-		}
-
-		// block Logger's io.Writer
-		w.mw.Lock()
```
Comment (JessonChan) [inline logs/file.go]: no need to lock here,we use a double-lock to guarantee this function be run thread-safe

File: logs/file.go [hunk 2]
```diff
@@ -210,50 +236,47 @@ func (w *fileLogWriter) lines() (int, error) {
...
-		err = w.startLogger()
-		if err != nil {
-			return fmt.Errorf("Rotate StartLogger: %s\n", err)
-		}
+	// Rename the file to its new found name
+	// even if occurs error,we MUST guarantee to  restart new logger
+	renameErr := os.Rename(w.Filename, fName)
+	// re-start logger
+	startLoggerErr := w.startLogger()
+	go w.deleteOldLog()
```
Comment (JessonChan) [inline logs/file.go]: even if occurs error,we MUST guarantee to  restart new logger and delete old logs.

File: logs/file.go [hunk 3]
```diff
@@ -150,10 +118,68 @@ func (w *fileLogWriter) WriteMsg(msg string, level int) error {
...
+	t = s / 10
+	buf[17] = byte('0' + t)
+	buf[18] = byte('0' + s - t*10)
+	buf[19] = ' '
+	msg = string(buf[0:]) + msg + "\n"
+
+	if w.Rotate {
+		if w.needRotate(len(msg), d) {
+			w.Lock()
+			if w.needRotate(len(msg), d) {
```
Comment (JessonChan) [inline logs/file.go]: double-cheked locking for thread-safe

File: logs/file.go [hunk 4]
```diff
@@ -122,27 +95,22 @@ func (w *fileLogWriter) Init(jsonconfig string) error {
...
 
-func (w *fileLogWriter) docheck(size int) {
-	w.startLock.Lock()
-	defer w.startLock.Unlock()
-	if w.Rotate && ((w.Maxlines > 0 && w.maxlinesCurlines >= w.Maxlines) ||
-		(w.Maxsize > 0 && w.maxsizeCursize >= w.Maxsize) ||
-		(w.Daily && time.Now().Day() != w.dailyOpendate)) {
-		if err := w.DoRotate(); err != nil {
-			fmt.Fprintf(os.Stderr, "FileLogWriter(%q): %s\n", w.Filename, err)
-			return
```
Comment (JessonChan) [inline logs/file.go]: 'return here' will create a bug:this line of log will be written but the current line number and current log size will not increase.

**Comment by JessonChan** [comment]
@astaxie @fuxiaohei

**Comment by astaxie** [comment]
LGTM

**Comment by fuxiaohei** [comment]
LGTM

**Pull request merged**